opendevin/server/agent/agent.py

```diff
@@ -175,10 +275,12 @@ def on_agent_event(self, event: Observation | Action):
         if isinstance(event, NullObservation):
             return
         event_dict = event.to_dict()
-        asyncio.create_task(self.send(event_dict),
-                            name='send event in callback')
+        if self.controller is not None:
+            event_dict['task_state'] = self.controller.get_task_state().value
+        self.send(event_dict)
```

Comment:
instead of sending this constantly, should we send an event when state changes?

Reply:
you are right. let me adjust it. btw, the `asyncio.create_task` to start the agent task will very likely block other threads. i think we should consider using multiprocessing to run the agent task in Python to realize the true async.

Reply:
agree!